src/main/scala/mesosphere/marathon/core/matcher/base/util/ActorOfferMatcher.scala

```diff
@@ -17,7 +18,7 @@ import scala.concurrent.{ Future, Promise }
   * @param actorRef Reference to actor that matches offers.
   * @param precedenceFor Defines which matcher receives offers first. See [[mesosphere.marathon.core.matcher.base.OfferMatcher.precedenceFor]].
   */
-class ActorOfferMatcher(actorRef: ActorRef, override val precedenceFor: Option[PathId])(implicit scheduler: akka.actor.Scheduler)
+class ActorOfferMatcher(actorRef: ActorRef, override val precedenceFor: Option[PathId], marathonScheduler: Provider[MarathonScheduler])(implicit scheduler: akka.actor.Scheduler)
```

Comment:
Do we really want to inject the scheduler or an abstraction? I have an uneasy feeling about the `MarathonScheduler` leaking into this code.

Reply:
by an abstraction you mean introducing some trait (something like HomeRegionProvider) and injecting that instead of the whole scheduler (although it will be implemented by the scheduler class)?

Reply:
I should have been more precise. It could be a trait or just a case class.